packages/tablev2/Table.tsx

```diff
@@ -231,30 +223,29 @@ export function Table<Entry, ColIDs extends string, Sort extends ColIDs>({
   const sort =
     col?.sorter?.(state.order, state.order === "asc" ? 1 : -1) ?? (() => 0);
   const sortedData = data.sort(sort);
+  const internalColumns: InternalColumn<Entry>[] = columns.map(c =>
+    Object.assign(c, { width: getWidth(c, state.columns) })
```

Comment:
this is the trick. we enhance the `Column` type by a width for internal use. that makes surrounding stuff less hard.